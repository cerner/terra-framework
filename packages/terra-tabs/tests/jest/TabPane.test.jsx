import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { shallowWithIntl } from 'terra-enzyme-intl';
import Pane from '../../src/TabPane';

jest.mock('uuid', () => ({ v4: () => '00000000-0000-0000-0000-000000000000' }));

describe('TabPane', () => {
  it('should render a default component with label', () => {
    const wrapper = shallowWithIntl(<Pane label="Default" />).dive().dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with icon and label', () => {
    const wrapper = shallowWithIntl(<Pane label="Label" icon={<div>Fake icon</div>} />).dive().dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with icon only when indicated', () => {
    const wrapper = shallowWithIntl(<Pane label="Label" icon={<div>Fake icon</div>} isIconOnly />).dive().dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a custom display when provided', () => {
    const wrapper = shallowWithIntl(<Pane label="Label" customDisplay={<div>Custom Display</div>} />).dive().dive();
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'orion-fusion-theme',
      });
    const wrapper = shallowWithIntl(<Pane label="Default" className="customClass" />).dive().dive();
    expect(wrapper).toMatchSnapshot();
  });
});
