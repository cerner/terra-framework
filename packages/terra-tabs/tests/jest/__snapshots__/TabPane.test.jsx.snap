// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPane correctly applies the theme context className 1`] = `
<div
  aria-disabled={false}
  aria-label="Default"
  aria-selected={false}
  className="tab is-text-only orion-fusion-theme"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Default"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-00000000-0000-0000-0000-000000000000"
      text="Terra.tabs.hint.removable"
    />
    <span
      className="label"
    >
      Default
    </span>
  </div>
</div>
`;

exports[`TabPane should render a default component with label 1`] = `
<div
  aria-disabled={false}
  aria-label="Default"
  aria-selected={false}
  className="tab is-text-only"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Default"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-00000000-0000-0000-0000-000000000000"
      text="Terra.tabs.hint.removable"
    />
    <span
      className="label"
    >
      Default
    </span>
  </div>
</div>
`;

exports[`TabPane should render with a custom display when provided 1`] = `
<div
  aria-disabled={false}
  aria-label="Label"
  aria-selected={false}
  className="tab is-text-only"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Label"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-00000000-0000-0000-0000-000000000000"
      text="Terra.tabs.hint.removable"
    />
    <div>
      Custom Display
    </div>
  </div>
</div>
`;

exports[`TabPane should render with icon and label 1`] = `
<div
  aria-disabled={false}
  aria-label="Label"
  aria-selected={false}
  className="tab"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Label"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-00000000-0000-0000-0000-000000000000"
      text="Terra.tabs.hint.removable"
    />
    <div>
      Fake icon
    </div>
    <span
      className="label"
    >
      Label
    </span>
  </div>
</div>
`;

exports[`TabPane should render with icon only when indicated 1`] = `
<div
  aria-disabled={false}
  aria-label="Label"
  aria-selected={false}
  className="tab is-icon-only"
  data-focus-styles-enabled={true}
  data-terra-tabs-show-focus-styles={true}
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="tab"
  style={
    Object {
      "zIndex": undefined,
    }
  }
  tabIndex={-1}
  title="Label"
>
  <div
    className="inner"
  >
    <VisuallyHiddenText
      aria-hidden={true}
      id="terra-tab-delete-pane-response-00000000-0000-0000-0000-000000000000"
      text="Terra.tabs.hint.removable"
    />
    <div>
      Fake icon
    </div>
  </div>
</div>
`;
